### ISSUE TITLE
task->name for tasks launched by 'needs' reports parent task name

### ISSUE THREAD

**Issue by shattered** [RexOps/Rex#1066]
How to demo with Rex 1.4:

```
task "a", sub { needs main ("b"); };
task "b", sub { run "id"; say sprintf("'b' is '%s'", task->name); };
```

Running 'a' will print 

> 'b' is 'a'

**Comment by krimdomu** [comment]
This will be fixed in the next release.

**Comment by alip** [comment]
This breaks `needs` for both `needs Pkg 'task'` and `needs 'task'` and reverting it restores the functionality. Am I missing something? Have you actually tested this?

**Comment by ferki** [comment]
TL;DR: `needs` seems to be only broken when `Pkg` contains Perl namespace separators. See #1188.

@alip: thanks for reporting! I've had some hard time to reproduce it first, but I believe I've finally found what might be broken for your use case. I've opened #1188 for double checking and further discussion.

Rex is being tested by its unit tests in this repo, and by its functional tests in [RexOps/rex-build](https://github.com/RexOps/rex-build) repo (which is executed against VMs running various OSes, to test if Rex can really manage them). There are plans to write more details about Rex testing when it is (re)integrated with Travis.
